src/test/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/CoverageVisualisationServiceTest.kt

```diff
@@ -0,0 +1,189 @@
+import com.intellij.openapi.wm.RegisterToolWindowTask
+import com.intellij.openapi.wm.ToolWindowAnchor
+import com.intellij.openapi.wm.ToolWindowManager
+import com.intellij.testFramework.fixtures.CodeInsightTestFixture
+import com.intellij.testFramework.fixtures.IdeaProjectTestFixture
+import com.intellij.testFramework.fixtures.IdeaTestFixtureFactory
+import com.intellij.testFramework.fixtures.JavaTestFixtureFactory
+import com.intellij.testFramework.fixtures.TestFixtureBuilder
+import nl.tudelft.ewi.se.ciselab.testgenie.settings.TestGenieSettingsService
+import org.assertj.core.api.Assertions.assertThat
+import org.evosuite.result.BranchInfo
+import org.evosuite.result.MutationInfo
+import org.evosuite.result.TestGenerationResultImpl
+import org.evosuite.shaded.org.mockito.Mockito
+import org.evosuite.utils.CompactReport
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CoverageVisualisationServiceTest {
+
+    private lateinit var fixture: CodeInsightTestFixture
+    private lateinit var coverageVisualisationService: CoverageVisualisationService
+    private lateinit var project: Project
+    private val editor = Mockito.mock(Editor::class.java)
+    private lateinit var coverageToolWindowDisplayService: CoverageToolWindowDisplayService
+
+    private val branch1 = Mockito.mock(BranchInfo::class.java)
+    private val branch2 = Mockito.mock(BranchInfo::class.java)
+    private val branch3 = Mockito.mock(BranchInfo::class.java)
+    private val branch4 = Mockito.mock(BranchInfo::class.java)
+    private val branch5 = Mockito.mock(BranchInfo::class.java)
+
+    private val mutation1 = Mockito.mock(MutationInfo::class.java)
+    private val mutation2 = Mockito.mock(MutationInfo::class.java)
+    private val mutation3 = Mockito.mock(MutationInfo::class.java)
+    private val mutation4 = Mockito.mock(MutationInfo::class.java)
+    private val mutation5 = Mockito.mock(MutationInfo::class.java)
+
+    @BeforeEach
+    fun setUp() {
+        val projectBuilder: TestFixtureBuilder<IdeaProjectTestFixture> =
+            IdeaTestFixtureFactory.getFixtureFactory().createFixtureBuilder("TestGenie")
+
+        fixture = JavaTestFixtureFactory.getFixtureFactory().createCodeInsightFixture(projectBuilder.fixture)
+        fixture.setUp()
+
+        project = fixture.project
+        coverageVisualisationService = project.service()
+        coverageToolWindowDisplayService = project.service()
+
+        // Initialise the toolWindow TestGenie
+        ToolWindowManager.getInstance(project)
+            .registerToolWindow(RegisterToolWindowTask("TestGenie", ToolWindowAnchor.RIGHT))
```

Comment:
Interesting, so you have to explicitly initialize our plugin?

Reply:
Yes, because I am making an empty project that does not have any plugins initialised in it. I have tried researching into it, but I could not find an alternative to this approach.